src/transformState.js

```diff
@@ -3,9 +3,35 @@
 /**
  * @param {Object} state
  * @param {Object[]} actions
- */
+*/
+
 function transformState(state, actions) {
-  // write code here
+  const stateModified = state;
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateModified, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateModified[key];
+        }
+        break;
+
+      case 'clear':
+        for (const key in stateModified) {
+          delete stateModified[key];
+        }
```

Comment:
Let's move this code into a separate function outside of our

Reply:
For now kept it inside the main function, but outside the for loop. I'm not reusing it somewhere else, so it seems fine. But tell me if it's a bad decision:pray: